src/app/public/modules/progress-indicator/progress-indicator.component.spec.ts

```diff
@@ -20,387 +20,649 @@ import {
+    progressIndicator = componentInstance.progressIndicator;
+
+    consoleWarnSpy = spyOn(console, 'warn');
   });
 
-  it('should not run the progressChanges emitter until a tick has occurred.', fakeAsync(() => {
-    fixture.detectChanges();
-    expect(fixture.componentInstance.progressChangesEmitted).toBeFalsy();
-    tick();
-    expect(fixture.componentInstance.progressChangesEmitted).toBeTruthy();
+  it('should set defaults', fakeAsync(function () {
+    detectChanges();
+
+    expect(progressIndicator.displayMode).toEqual(SkyProgressIndicatorDisplayMode.Vertical);
+    expect(progressIndicator.isPassive).toEqual(false);
+    expect(progressIndicator.startingIndex).toEqual(0);
   }));
 
-  it('should use horizontal display if set', fakeAsync(() => {
-    fixture.componentInstance.displayMode = SkyProgressIndicatorDisplayMode.Horizontal;
-    fixture.detectChanges();
-    tick();
-    fixture.detectChanges();
-    componentInstance = fixture.componentInstance.progressIndicator;
-    let element = fixture.nativeElement;
+  it('should emit progress changes initially', fakeAsync(function () {
+    const spy = spyOn(componentInstance, 'onProgressChanges').and.callThrough();
 
-    expect(componentInstance.isHorizontal).toBeTruthy();
-    for (let item of componentInstance.progressItems.toArray()) {
-      expect(item.isHorizontal).toBeTruthy();
-    }
+    detectChanges();
 
-    expect(element.querySelector('.sky-progress-indicator-display')).toBeTruthy();
-    expect(element.querySelector('.sky-progress-indicator-item-step')).toBeFalsy();
+    expect(spy).toHaveBeenCalled();
   }));
 
-  it('should use passive mode if set', fakeAsync(() => {
-    fixture.componentInstance.isPassive = true;
-    fixture.detectChanges();
-    tick();
-    fixture.detectChanges();
-    componentInstance = fixture.componentInstance.progressIndicator;
+  it('should use horizontal display if set', fakeAsync(function () {
+    componentInstance.displayMode = SkyProgressIndicatorDisplayMode.Horizontal;
+
+    detectChanges();
 
-    expect(componentInstance.isPassive).toBeTruthy();
-    for (let item of componentInstance.progressItems.toArray()) {
-      expect(item.isPassive).toBeTruthy();
-    }
+    const element = fixture.nativeElement;
+
+    expect(element.querySelector('.sky-progress-indicator-status-markers')).toBeTruthy();
+    expect(element.querySelector('.sky-progress-indicator-item .sky-progress-indicator-status-marker')).toBeFalsy();
   }));
 
-  // May be removed in the future if support is added
-  it('should not use passive mode if set for horizontal display', fakeAsync(() => {
-    fixture.componentInstance.displayMode = SkyProgressIndicatorDisplayMode.Horizontal;
-    fixture.componentInstance.isPassive = true;
-    fixture.detectChanges();
-    tick();
-    fixture.detectChanges();
-    componentInstance = fixture.componentInstance.progressIndicator;
+  it('should use starting index if set', fakeAsync(function () {
```

Comment:
I added a comment above the expectation to this effect (to avoid making the `it` message too long).

Reply:
Would you mind also adding something to the title like "... and ensure previous steps are marked complete" ? I think its important to note this type of business logic. (as I'm sure I'll forget in the future).